src/lib.js

```diff
@@ -1333,6 +1346,31 @@ export const getItems = async (
     output.armor_set_rarity = constants.RARITIES[Math.max(...armor.map((a) => helper.rarityNameToInt(a.rarity)))];
   }
 
+  // Full equipment set (4 pieces)
+  for (const piece of equipment) {
+    if (piece.rarity == null) delete equipment[equipment.indexOf(piece)];
```

Comment:
It feels like we would be better off just filtering in the first place. I think I have seen this 3 times now